niceman/resource/session.py

```diff
@@ -195,18 +197,16 @@ def niceman_exec(self, command, args):
     @abc.abstractmethod
     def exists(self, path):
         """Return if file exists"""
-        pass
+        raise NotImplementedError
 
     @abc.abstractmethod
-    def put(self, src_path, dest_path, preserve_perms=False,
-                owner=None, group=None, recursive=False):
+    def put(self, src_path, dest_path, owner=None, group=None):
```

Comment:
So will all of them operate rrcursively?

Reply:
Yes, the gets and puts in the Starcluster package are magically recursive.